src/main/java/org/kitteh/irc/client/library/feature/InteractionDefaults.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * * Copyright (C) 2013-2016 Matt Baxter http://kitteh.org
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.feature;
+
+import org.kitteh.irc.client.library.util.Sanity;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InteractionDefaults implements InteractionMap {
+    protected final Map<Interaction, String> defaults = new HashMap<>();
+    protected @Nullable final String defaultString;
+
+    /**
+     * Creates a InteractionDefaults object with a custom string to default
+     * to when an Interaction key is not set.
+     *
+     * Will throw an IllegalArgumentException is you set the
+     * defaultString value to "". Use null instead.
+     *
+     * @param defaultString a default string to use for when a key is not set
+     */
+    public InteractionDefaults(@Nullable String defaultString) {
+        this.emptyStringCheck(defaultString);
+
+        this.defaultString = defaultString;
+    }
+
+    /**
+     * Creates a InteractionDefaults object with predefined string to default
+     * to when an Interaction key is not set.
+     */
+    public InteractionDefaults() {
+        this.defaultString = null;
+    }
+
+    /**
+     * Sets a default message for the Interaction.
+     *
+     * Will throw an IllegalArgumentException is you set the
```

Comment:
KICL has many cases of 

```java
return "KICK zarthus #zarthus " + (message != null ? ": " + message : "");
```

If you want the string to be nothing, use null. This prevents KICL from sending useless data, gives you only one value to expect instead of 2. It's explained in the comment for emptyStringCheck.